geode-core/src/main/java/org/apache/geode/management/internal/operations/OperationManager.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.management.internal.operations;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+import org.apache.geode.internal.logging.LoggingExecutors;
+import org.apache.geode.management.operation.OperationResult;
+
+/**
+ * Maintains a list of known, possible operations and can run them when requested. Maintains state
+ * of running operations
+ */
+public class OperationManager {
+
+  private final Map<String, OperationExecutor> operations = new HashMap<>();
+
+  private final Map<String, OperationResult> operationResults = new HashMap<>();
+
+  public void registerOperation(OperationExecutor executor) {
+    operations.put(executor.getId(), executor);
```

Comment:
Should it be an error if the id is already in the map or did you just want to silently replace it?